Library/Homebrew/utils/curl.rb

```diff
@@ -94,7 +94,7 @@ def curl_check_http_content(url, user_agents: [:default], check_content: false,
   user_agents.each do |ua|
     details = curl_http_content_headers_and_checksum(url, hash_needed: hash_needed, user_agent: ua)
     user_agent = ua
-    break if details[:status].to_s.start_with?("2")
+    break if details[:status].start_with?("1", "2")
```

Comment:
Don't drop the `to_s` here (or below); it's needed for cases where `details[:status]` isn't set. Alternatively, use `&`

Could also consider consolidating all these into a method `status_ok?(status)` that unifies this logic between all the methods?

Reply:
`(100..299).include?(details[:status].to_i)`

Reply:
I think I'll refrain from adding a method for now; if we end up needing the logic in more places, I do think it would be warranted though.

Reply:
@alebcay It's used in four places so a method would be good if you don't mind 😄